sdk/formrecognizer/Azure.AI.FormRecognizer/src/TrainingOperation.cs

```diff
@@ -96,16 +114,41 @@ private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToke
                     ? await _serviceClient.GetCustomModelAsync(new Guid(Id), includeKeys: true, cancellationToken).ConfigureAwait(false)
                     : _serviceClient.GetCustomModel(new Guid(Id), includeKeys: true, cancellationToken);
 
-                if (update.Value.ModelInfo.Status != CustomFormModelStatus.Training)
+                _response = update.GetRawResponse();
+
+                if (update.Value.ModelInfo.Status == CustomFormModelStatus.Ready)
                 {
                     _hasCompleted = true;
                     _value = new CustomFormModel(update.Value);
                 }
+                else if (update.Value.ModelInfo.Status == CustomFormModelStatus.Invalid)
+                {
+                    _hasCompleted = true;
+                    _modelId = update.Value.ModelInfo.ModelId;
 
-                _response = update.GetRawResponse();
+                    throw await CreateExceptionForFailedOperationAsync(async, update.Value.TrainResult.Errors).ConfigureAwait(false);
+                }
             }
 
             return GetRawResponse();
         }
+
+        private async ValueTask<RequestFailedException> CreateExceptionForFailedOperationAsync(bool async, IReadOnlyList<FormRecognizerError> errors)
+        {
+            string errorMessage = $"Invalid model created with ID {_modelId}";
```

Comment:
>Is this because we want to create our own exception here with the modelId?

Correct. We want to let the user know the modelid so they can access the model and take action on it.
As an example of error, if the bloburl provided is wrong, the message will say something like "Invalid container path" but a model was still created